### ISSUE TITLE
[BUG] flatseasons bug

### ISSUE THREAD

**Issue by roidy** [jurialmunkey/plugin.video.themoviedb.helper#1232]
### Describe the bug

I have the addon set to flatten seasons and have found a bug.

If I open info dialog on a show e.g. Game of Thrones then `ListItem.FolderPath` correctly returns:-

`plugin://plugin.video.themoviedb.helper/?info=flatseasons&tmdb_type=tv&tmdb_id=1399`

However if I now open the recommendations dialog and select a show from a list like recommendations e.g. the show Breaking Bad then `ListItem.FolderPath` for the new show returns:-

`plugin://plugin.video.themoviedb.helper/?info=seasons&tmdb_type=tv&tmdb_id=1396`

It is no longer returning `flatseasons`, then if I go back to the original starting info dialog for Game of Thrones that originally correctly returned `flatseasons` it now returns just `seasons`:-

`plugin://plugin.video.themoviedb.helper/?info=seasons&tmdb_type=tv&tmdb_id=1399`

### Steps To Reproduce

While logging `ListItem.FolderPath` to a label do the following.

1. Set the addon to use Navigation->Flatten seasons
2. Open info on a TV Show from the addon:- `ListItem.FolderPath` correctly contains `info=flatseasons`
3. Open the recommendations dialog and then open info on a new TV Show from a list like similar or recommendations:- `ListItem.FolderPath` incorrectly contains `info=seasons`
4. Now navigate back to the original info dialog in step 1 that was correct and on return `ListItem.FolderPath` will now incorrectly contain `info=seasons`

### Checklist

- [X] I have searched the issues and this bug has not been reported.
- [X] I have updated TMDbHelper to the most recent version.
- [X] I have described the bug and included steps to reproduce the issue
- [ ] I have uploaded my debug log.

**Comment by jurialmunkey** [comment]
Wow, this was really doing my head in. Was looking all over the recommendations script trying to figure out how it could possibly be breaking the path. Turned out to be a complete red herring. 

The real problem was that I never even implemented the flatseasons setting properly for the detailed item directory. It was broken in a lot of places, not just recs. Worthwhile in the end and a super simple fix, but omg I was tearing my hair out until I realised!
